lib/iris/analysis/trajectory.py

```diff
@@ -341,6 +341,8 @@ def interpolate(cube, sample_points, method=None):
             # This is **not** proper mask handling, because we cannot produce a
             # masked result, but it ensures we use a "filled" version of the
             # input in this case.
+            if cube.fill_value is not None:
+                source_data.fill_value = cube.fill_value
```

Comment:
The way `trajectory` handles masks is pretty ropey ... this change appears appropriate.